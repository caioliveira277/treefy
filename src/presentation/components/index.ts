export * from './button/button-component';
export * from './text-input/text-input-component';
export * from './link/link-component';
export * from './legend/legend-component';
export * from './picker-input/picker-input-component';
export * from './page-loading/page-loading-component';
export * from './toast-message/toast-message-component';
export * from './skeleton-loading/skeleton-loading-component';
export * from './empty-content/empty-content-component';
export * from './search-input/search-input-component';
export * from './animated-height/animated-height';
export * from './loading-overlay/loading-overlay-component';
