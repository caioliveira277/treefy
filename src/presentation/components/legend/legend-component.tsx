import { StyleProp, ViewStyle } from 'react-native';
import { Legend } from './styles';

export interface LegendComponentProps {
  children: string;
  style?: StyleProp<ViewStyle>;
}

export const LegendComponent: React.FC<LegendComponentProps> = ({
  children,
  style,
}) => <Legend style={style}>{children}</Legend>;
