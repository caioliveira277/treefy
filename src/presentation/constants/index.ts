export const TREEFY_SUPPORT = {
  email: 'treefy.suporte@gmail.com',
  cel1: '(15) 99782-7880',
  cel2: '(11) 93203-7301',
};
