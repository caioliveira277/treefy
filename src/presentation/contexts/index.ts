export * from './authentication-context';
export * from './introduction-context';
export * from './toast-context';
