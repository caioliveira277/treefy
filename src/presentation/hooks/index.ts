export * from './use-debounce/use-debounce-hook';
