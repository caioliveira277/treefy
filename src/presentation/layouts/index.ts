export * from './public/public-layout';
export * from './profile/profile-layout';
