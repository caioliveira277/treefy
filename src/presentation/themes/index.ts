import { primaryTheme } from './primary-theme';
export const currentTheme = (() => primaryTheme)();
