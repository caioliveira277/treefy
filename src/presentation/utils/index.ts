export * from './get-icons/get-icons-utils';
export * from './format-date/format-date-utils';
