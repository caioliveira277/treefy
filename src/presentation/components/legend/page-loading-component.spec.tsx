import { mockRender } from '@/presentation/tests';
import { LegendComponent, LegendComponentProps } from './legend-component';
import { RenderAPI } from '@testing-library/react-native';

type SutTypes = {
  sut: RenderAPI;
};
interface MakeSutProps extends Partial<LegendComponentProps> {}

const makeSut = ({ children = 'Any content' }: MakeSutProps): SutTypes => {
  const sut = mockRender(<LegendComponent>{children}</LegendComponent>);
  return {
    sut,
  };
};

describe('LegendComponent', () => {
  test('Should match with a snapshot', () => {
    const { sut } = makeSut({});
    const { toJSON } = sut;
    expect(toJSON()).toMatchSnapshot();
  });
});
