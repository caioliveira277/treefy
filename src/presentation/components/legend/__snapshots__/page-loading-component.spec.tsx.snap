// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendComponent Should match with a snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#585960",
        "fontFamily": "RobotoSlab_400Regular",
        "fontSize": 12,
        "marginBottom": 10,
      },
    ]
  }
  theme={
    Object {
      "borders": Object {
        "border_radius_md": "10px",
        "border_radius_sm": "5px",
      },
      "colors": Object {
        "body": "#585960",
        "complete": "#4ABC86",
        "contrast": "#FCFDFF",
        "edit": "#4ABCBC",
        "error": "#E04244",
        "info": "#01A0C4",
        "placeholder": "#C6C6C6",
        "placeholder_light": "#EEEEEE",
        "primary": "#4ABC86",
        "rate_active": "#D56D6D",
        "rate_inactive": "#E5A6A7",
        "secondary": "#253239",
        "success": "#79AA05",
        "sun": "#F0DE5F",
        "warning": "#C46403",
        "water": "#B0D8FE",
        "white": "#ffffff",
      },
      "fonts": Object {
        "families": Object {
          "bold": "RobotoSlab_700Bold",
          "medium": "RobotoSlab_500Medium",
          "regular": "RobotoSlab_400Regular",
        },
        "sizes": Object {
          "lg": "16px",
          "md": "14px",
          "sm": "12px",
          "xl": "18px",
        },
      },
      "shadows": Object {
        "sm": Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2.62,
        },
      },
    }
  }
>
  Any content
</Text>
`;
